### PR TITLE
feat: add TransferWise to Connect Account menu

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/2972

# Description
Add TransferWise as an option in Connected Accounts settings menu.

This is being guarded by a feature flag, in order to test make sure you set `settings.features.transferwise = true` in your collective.

Requires https://github.com/opencollective/opencollective-api/pull/3532

### CODE REVIEW

**Comment by kewitz** [comment]
Rebased.

File: components/EditConnectedAccount.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { HelpBlock, Button } from 'react-bootstrap';
```
Comment (Betree) [inline components/EditConnectedAccount.js]: :rocket:

File: components/edit-collective/EditTransferWiseAccount.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      id
+    }
+  }
+`;
+
+const mutationOptions = { context: API_V2_CONTEXT };
+
+const EditTransferWiseAccount = props => {
+  const messages = defineMessages({
+    'collective.connectedAccounts.transferwise.button': {
```
Comment (Betree) [inline components/edit-collective/EditTransferWiseAccount.js]: **[:ice_cream: Details & personal preferences]** I personally prefer to use simple keys for defined message objects, something like:

```suggestion
    button: {
```

That way you can simply do `formatMessage(messages.button)` to translate the string.

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: components/edit-collective/EditTransferWiseAccount.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      token: '',
+    },
+    async onSubmit(values) {
+      const { data } = await createConnectedAccount({
+        variables: {
+          connectedAccount: { token: values.token, service: 'transferwise' },
+          account: { slug: props.collective.slug },
+        },
+      });
+      console.log(data);
```
Comment (Betree) [inline components/edit-collective/EditTransferWiseAccount.js]:
```suggestion
```

Comment (kewitz) [reply components/edit-collective/EditTransferWiseAccount.js]: 🚨!

**Comment by kewitz** [comment]
Thanks for the review, I forgot to attach the gif:
Kapture 2020-03-18 at 17 02 41

**Comment by kewitz** [comment]
@Betree, I just noticed the use of state is bad because the user can go to another tab and come back again.

The whole ConnectedAccount tree of components use state, which is causing the refetch to not update the DOM.

I'll get rid of this tree in favor of pure components so everything is always up-to-date with the DB.

File: components/edit-collective/EditTransferWiseAccount.js (hunk 2 above)
Comment (kewitz) [reply components/edit-collective/EditTransferWiseAccount.js]: Nice, I'll start doing that.

**Comment by kewitz** [comment]
Kapture 2020-03-20 at 10 18 38

**Comment by Betree** [comment]
> @Betree, I just noticed the use of state is bad because the user can go to another tab and come back again.
> 
> The whole ConnectedAccount tree of components use state, which is causing the refetch to not update the DOM.
> 
> I'll get rid of this tree in favor of pure components so everything is always up-to-date with the DB.

This one is a known issue: https://github.com/opencollective/opencollective/issues/1451

Your solution is ok, the extra mile would be to also fetch the data (`connectedAccounts`) from the `EditConnectedAccounts` rather than relying on the props.

**Pull request merged**